src/group_theory/perm/basic.lean

```diff
@@ -196,6 +196,18 @@ extend_domain_refl f
   (e.extend_domain f) * (e'.extend_domain f) = (e * e').extend_domain f :=
 extend_domain_trans _ _ _
 
+/-- `extend_domain` as a group homomorphism -/
+def extend_domain_hom : perm α →* perm β :=
+{ to_fun := λ e, extend_domain e f,
+  map_one' := extend_domain_one f,
+  map_mul' := λ e e', (extend_domain_mul f e e').symm }
+
+lemma extend_domain_hom_apply : extend_domain_hom f e = extend_domain e f := rfl
```

Comment:
```suggestion
/-- `extend_domain` as a group homomorphism -/
@[simps]
def extend_domain_hom : perm α →* perm β :=
{ to_fun := λ e, extend_domain e f,
  map_one' := extend_domain_one f,
  map_mul' := λ e e', (extend_domain_mul f e e').symm }
```

Reply:
Sure, but what does `simps` do?

Reply:
It generates `extend_domain_hom_apply` for you, and marks it with `@[simp]`